lib/SILOptimizer/Mandatory/DefiniteInitialization.cpp

```diff
@@ -1079,7 +1079,7 @@ void LifetimeChecker::handleInOutUse(const DIMemoryUse &Use) {
             // Normal method calls are curried, so they are:
             // (call_expr (dot_syntax_call_expr (decl_ref_expr METHOD)))
             FD = dyn_cast<FuncDecl>(DSCE->getCalledValue());
-          else
+          else if (CE->getCalledValue() != nullptr)
```

Comment:
Another possibility is to use dyn_cast_or_null.

http://llvm.org/docs/ProgrammersManual.html#the-isa-cast-and-dyn-cast-templates